manifests/plugin/pcie_errors.pp

```diff
@@ -0,0 +1,23 @@
+# Class to manage pcie_errors plugin for collectd
+#
+# @param ensure Ensure param for collectd::plugin type.
+# @param source Use sysfs or proc to read data from /sysfs or /proc.
+# @param access_dir Directory used to access device config space.
+# @param report_masked If true plugin will notify about errors that are set to masked in Error Mask register.
+# @param persistent_notifications If false plugin will dispatch notification only on set/clear of error.
+#
+class collectd::plugin::pcie_errors (
+  Enum['present', 'absent'] $ensure                   = 'present',
+  Optional[String]          $source                   = 'sysfs',
```

Comment:
We are never able to set $source to undef if it has a default value. We can only change `sysfs`  to another string. Are sysfs and proc the only valid values? In that case we could use an enum here?

Reply:
Good point. Values other than sysfs and proc are treated as error. As suggested I changed data type to enum and updated test case.